src/main/java/ecdar/Ecdar.java

```diff
@@ -299,27 +299,16 @@ public void start(final Stage stage) {
 
         stage.setOnCloseRequest(event -> {
             BackendHelper.stopQueries();
-
-            try {
-                backendDriver.closeAllEngineConnections();
-            } catch (IOException e) {
-                e.printStackTrace();
-            }
+            backendDriver.closeAllEngineConnections();
```

Comment:
Could this be replaced with a  ``reset`` call like at ``addEngineInstanceListener``?

Reply:
The idea was that we could avoid clearing the connection lists, as these would be freed automatically on shutdown, and we would only need to close connections (gRPC channels and running processes). But it is very cheap in most cases (there are mostly few elements in them, especially with multi-threaded engines). I will change it